src/tools/CMakeLists.txt

```diff
@@ -0,0 +1,261 @@
+
+# ~~~ Find libraries  
+
+# ~ Iconv
+if(UNIX)
+    find_package(Iconv REQUIRED)
+endif()
+
+# ~ Curses
+if(UNIX)
+    option(HUNSPELL_TOOLS_USE_CURSES  "Locate and link curses library."  OFF)
+
+    if(HUNSPELL_TOOLS_USE_CURSES)
+        set(CURSES_NEED_WIDE ON)
+        find_package(Curses)
+
+        if(NOT CURSES_FOUND)
+            message(FATAL_ERROR  "Could not find curses library. Set paths manually.")
+        endif()
+    endif()
+endif()
+
+
+if(CURSES_FOUND)
+    set(_HAVE_CURSES_H    OFF)
+    set(_HAVE_NCURSESW_H  OFF)
+    set(_CURSES_INC_DIR   "")
+    set(_CURSES_LIB       "")
+
+    get_filename_component(_CURSES_LIB_NAME  ${CURSES_NCURSES_LIBRARY}  NAME)
+
+    # libcurses just points to libncurses in most cases
+    if(EXISTS "${CURSES_INCLUDE_PATH}/curses.h")
+        set(_HAVE_CURSES_H   ON)
+        set(_CURSES_INC_DIR  "${CURSES_INCLUDE_PATH}")
+        set(_CURSES_LIB      "${CURSES_CURSES_LIBRARY}")
+    endif()
+    if(EXISTS "${CURSES_INCLUDE_PATH}/ncurses/curses.h")
+        set(_HAVE_CURSES_H   ON)
+        set(_CURSES_INC_DIR  "${CURSES_INCLUDE_PATH}/ncurses")
+        set(_CURSES_LIB      "${CURSES_NCURSES_LIBRARY}")
+    endif()
+    if(EXISTS "${CURSES_INCLUDE_PATH}/ncursesw/curses.h" AND ${_CURSES_LIB_NAME} MATCHES "w")
+        set(_HAVE_NCURSESW_H  ON)
+        set(_CURSES_INC_DIR   "${CURSES_INCLUDE_PATH}")
+        set(_CURSES_LIB       "${CURSES_NCURSES_LIBRARY}")
+    endif()
+endif()
+
+# ~ Readline
+if(UNIX)
+    option(HUNSPELL_TOOLS_USE_READLINE  "Locate and link readline library."  OFF)
+endif()
+
+if(HUNSPELL_TOOLS_USE_READLINE)
+    find_path(READLINE_INCLUDE_PATH
+        NAMES  readline/readline.h
+        DOC    "Readline include directory."
+    )
+    find_library(READLINE_LIBRARY
+        NAMES  readline
+        DOC    "Readline library."
+    )
+    mark_as_advanced(
+        READLINE_INCLUDE_PATH
+        READLINE_LIBRARY
+    )
+    find_package_handle_standard_args(Readline
+        REQUIRED_VARS  
+            READLINE_LIBRARY
+            READLINE_INCLUDE_PATH
+    )
+    if(NOT Readline_FOUND)
+        message(ERROR  "Could not find readline library. Set paths manually.")
+    endif()
+endif()
+
+# ~~~ Define VERSION in a dummy config.h header file, other defs are added 
+#     through target_compile_definitions()
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/config.h.in [=[#define VERSION "@VERSION@"]=])
+configure_file(${CMAKE_CURRENT_BINARY_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h @ONLY)
+
+# ~~~ Update target
+target_compile_definitions(${HUNSPELL_TOOLS_HUNSPELL_TARGET}
+    PRIVATE
+        $<$<BOOL:${_HAVE_LOCALE_H}>:HAVE_LOCALE_H>
+        $<$<BOOL:${_HAVE_LANGINFO_H}>:HAVE_LANGINFO_H>
+        $<$<BOOL:${Iconv_FOUND}>:HAVE_ICONV>
+        $<$<BOOL:${Iconv_FOUND}>:ICONV_CONST=>
+        $<$<BOOL:${_HAVE_CURSES_H}>:HAVE_CURSES_H>
+        $<$<BOOL:${_HAVE_NCURSESW_H}>:HAVE_NCURSESW_CURSES_H>
+        $<$<BOOL:${Readline_FOUND}>:HAVE_READLINE>
```

Comment:
These are known at configure time, please do not defer them to generation time using genexes.